example/object.h

```diff
@@ -55,37 +62,43 @@ class Object {
 template <typename T> class ref {
 public:
 	/// Create a nullptr reference
-    ref() : m_ptr(nullptr) { std::cout << "Created empty ref" << std::endl; }
+    ref() : m_ptr(nullptr) { print_default_created(this); track_default_created((ref_tag*) this); }
```

Comment:
No, you're right, this is exposing a bit more.  I thought that might be useful, seeing how this is serving as an example of a custom smart pointer.  The specific pointer values should probably be preserved in the output, however, as they do make it easier to follow.  I'll add them.